packages/compass-query-bar/src/components/query-history/query-item/query-item-action-buttons.tsx

```diff
@@ -1,54 +1,85 @@
-import { Icon, IconButton } from '@mongodb-js/compass-components';
+import { Icon, IconButton, css } from '@mongodb-js/compass-components';
 import React from 'react';
 
 type ActionButtonProps = {
-  onClick: () => void;
+  onClick?: () => void;
+  visible?: boolean;
 };
 
-export const FavoriteActionButton = ({ onClick }: ActionButtonProps) => {
+const VISIBLE_DISPLAY = 'inline';
+const INVISIBLE_DISPLAY = 'none';
+
+const exhaustIfInteractive =
+  (onClick: (() => void) | undefined) => (event: React.MouseEvent) => {
+    if (onClick) {
+      event.stopPropagation();
+      onClick?.();
+    }
+  };
+
+export const FavoriteActionButton = ({
+  onClick,
+  visible,
+}: ActionButtonProps) => {
   return (
     <IconButton
+      className={css({
```

Comment:
Let's not use `css` in render, this have perf implications

Also if you need to remove the button completely, why not just render nothing here? (Or from the parent component as this seems to be added to handle the "hovered" state that parent controls)